pandas/core/frame.py

```diff
@@ -665,6 +666,17 @@ def __init__(
                 NDFrame.__init__(self, data)
                 return
 
+        try:
```

Comment:
I'd prefer pointing user to ``to_pandas`` for now. This solution does not respect ``pandas_metadata``, which seems unfortunate. Also this adds quite significant overhead when arrow is not installed:

```
%timeit pd.DataFrame(arr)
86.6 µs ± 23.6 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

%timeit pd.DataFrame(arr)
7.49 µs ± 28.8 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

I'd prefer something like pd.DataFrame.from_arrow or similar

Reply:
Interesting finding. Not sure if necessarily a problem, since the overhead is constant, it doesn't grow with the size or type of data being loaded, seems to be a constant 80 microseconds (in my machine it take the same when pyarrow is not installed, but double than you when it is). I understand the Python interpreter looks in more places when the module is not immediately found, but that's like 6 or 12 times more than when it's found.

In absolute values 80 microseconds constant is not something that seems that bad, but since it's 10 times more than a normal use case, seems like a fair point to avoid. At least until PyArrow is a required dependency (or never, since we probably should deprecate having a constructor that loads everything in favor of a factory approach).

For the metadata, can you clarify which is the metadata not being respected?

Reply:
The following should preserve the index:

```
df = DataFrame({"a": [1, 2, 3]}, index=[1, 2, 3])
table = pyarrow.table(df)
DataFrame(table)
```

this returns 

```   a  __index_level_0__
0  1                  1
1  2                  2
2  3                  3
```

to_pandas does this automatically, we have the same problem in read_parquet and friends, that's why I put up #51766

Reply:
Yep, very good point. I didn't think about the index, I see there is some more logic in the transformation: https://github.com/apache/arrow/blob/68e89ac9ee5bf2c0c68493eb9405554162fdfab8/python/pyarrow/pandas_compat.py#L797

I guess it doesn't makes sense to replicate all that logic in our code, since PyArrow already has it. I guess `DataFrame.from_arrow` as a wrapper to `Table.to_pandas` (using the same parameters) is what make more sense, no?

Reply:
Yep agreed. Users can use set_index and rename if they want to add an index or rename columns anyway, so no real need for the constructor functionality

Reply:
Now that I think about it, I'm not sure if it's better to use `from_arrow` or `from_pyarrow`. My bet is that it's likely that another Arrow implementation in Python but based in Arrow can be developed eventually. Polars must have reimplemented PyArrow in a way, and whether it's because that code is moved of Polars, or another project is created, I guess it'll end up happening.

Maybe I'm overthinking, but at that point, is it more natural to have something like `DataFrame.from_pyarrow` and `DataFrame.from_arrow2`, or just one `DataFrame.from_arrow` that supports both? Small preference for the former, but happy to hear other opinions.